File: wayland/src/main/java/org/westmalle/wayland/nativ/xcb_expose_event_t.java
//Copyright 2015 Erik De Rijcke
//
//Licensed under the Apache License,Version2.0(the"License");
//you may not use this file except in compliance with the License.
//You may obtain a copy of the License at
//
//http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,software
//distributed under the License is distributed on an"AS IS"BASIS,
//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express or implied.
//See the License for the specific language governing permissions and
//limitations under the License.
package org.westmalle.wayland.nativ;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

public class xcb_expose_event_t extends Structure {

    private static final List<?> FIELD_ORDER = Arrays.asList("response_type",
                                                             "pad0",
                                                             "sequence",
                                                             "window",
                                                             "x",
                                                             "y",
                                                             "width",
                                                             "height",
                                                             "count");
    public byte  response_type;
    public byte  pad0;
    public short sequence;
    public int   window;
    public short x;
    public short y;
    public short width;
    public short height;
    public short count;

    public xcb_expose_event_t(final Pointer p) {
        super(p);
    }

    @Override
    protected List<?> getFieldOrder() {
        return FIELD_ORDER;
    }
}

File: compositor/src/main/kotlin/org/westford/nativ/libxcb/xcb_expose_event_t.kt
/*
 * Westford Wayland Compositor.
 * Copyright (C) 2016  Erik De Rijcke
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.westford.nativ.libxcb

import org.freedesktop.jaccall.CType
import org.freedesktop.jaccall.Field
import org.freedesktop.jaccall.Struct

@Struct(Field(name = "response_type",
              type = CType.CHAR),
        Field(name = "pad0",
              type = CType.CHAR),
        Field(name = "sequence",
              type = CType.SHORT),
        Field(name = "window",
              type = CType.INT),
        Field(name = "x",
              type = CType.SHORT),
        Field(name = "y",
              type = CType.SHORT),
        Field(name = "width",
              type = CType.SHORT),
        Field(name = "height",
              type = CType.SHORT),
        Field(name = "count",
              type = CType.SHORT)) class xcb_expose_event_t : Struct_xcb_expose_event_t()
